### PR TITLE
Spanner docs updates and fixes

### PR DESCRIPTION
**Pull request by quartzmo**
This PR adds spanner to the docs manifest, updates the TOC listing, and makes various updates and fixes. It also adds the `true` return value to some `Session` methods.

### CODE REVIEW

File: google-cloud-spanner/docs/toc.json [hunk 1]
```diff
@@ -32,26 +32,38 @@
...
           "type": "google/cloud/spanner/v1",
-          "patterns": ["\/spanner\/v1", "\/spanner\/admin\/instance\/v1", "\/spanner\/admin\/database\/v1", "google\/iam", "google\/protobuf", "google\/rpc"],
+          "patterns": ["\/spanner\/v1", "google\/iam", "google\/protobuf", "google\/rpc"],
           "nav": [
             {
               "title": "SpannerClient",
               "type": "google/cloud/spanner/v1/spannerclient"
             },
             {
-              "title": "InstanceAdminClient",
```
Comment (blowmage) [inline google-cloud-spanner/docs/toc.json]: Why were these removed? Spanner has 3 APIs, I think it would be good to show all 3.

File: google-cloud-spanner/lib/google/cloud/spanner/commit.rb [hunk 2]
```diff
@@ -127,6 +129,8 @@ def upsert table, *rows
         #   See [Data
         #   types](https://cloud.google.com/spanner/docs/data-definition-language#data_types).
         #
+        # @return [Array<Hash>] The array of rows.
```
Comment (blowmage) [inline google-cloud-spanner/lib/google/cloud/spanner/commit.rb]: Why is `upsert` returning the rows? I don't think it should be returning anything.

Comment (quartzmo) [inline google-cloud-spanner/lib/google/cloud/spanner/commit.rb]: I don't know why, but that is what the method returns.

File: google-cloud-spanner/docs/toc.json (hunk 1 above)
Comment (quartzmo) [inline google-cloud-spanner/docs/toc.json]: Right, I'll restore those other GAPIC clients, thanks.

File: google-cloud-spanner/lib/google/cloud/spanner/commit.rb (hunk 2 above)
Comment (blowmage) [inline google-cloud-spanner/lib/google/cloud/spanner/commit.rb]: I think it would be better to leave that behavior undocumented. It's not part of the API design.

Comment (quartzmo) [inline google-cloud-spanner/lib/google/cloud/spanner/commit.rb]: Done, removed.
